### DISCUSSION TITLE
Github action job doesn't fail on exit status 1

### DISCUSSION THREAD

**Discussion by TheOnlyWei** [community/community#26737] [Actions]
<p>I have a Golang code that runs some integration tests. When it exits with status 1, the Github Actions job is still successful and continues running. Should it not fail the job? What determines if the job fails in Github Actions?</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/4/3/4324cd9f1e281b8c5aaaef67a91d80cb0501445a.png" data-download-href="/uploads/short-url/9zYQ8VvmVvnRPOozbERLTS5WRey.png?dl=1" title="Screenshot 2021-12-02 144954" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/19400755/181099665-9a7fa6a3-2260-4352-abea-07dd89a8e07d.png" alt="Screenshot 2021-12-02 144954" data-base62-sha1="9zYQ8VvmVvnRPOozbERLTS5WRey" width="690" height="156" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/4/3/4324cd9f1e281b8c5aaaef67a91d80cb0501445a_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">Screenshot 2021-12-02 144954</span><span class="informations">1105×250 8.12 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>
<p>Here is the link to the code:</p><aside class="onebox githubblob" data-onebox-src="https://github.com/Azure-Samples/Hybrid-Golang-Samples/blob/0fd65d23d5efd4fb6b5a3f5b026f69a842af9d88/dataplane/app.go#L126">
  <header class="source">

      <a href="https://github.com/Azure-Samples/Hybrid-Golang-Samples/blob/0fd65d23d5efd4fb6b5a3f5b026f69a842af9d88/dataplane/app.go#L126" target="_blank" rel="noopener nofollow ugc">github.com</a>
  </header>

  <article class="onebox-body">
    <h4><a href="https://github.com/Azure-Samples/Hybrid-Golang-Samples/blob/0fd65d23d5efd4fb6b5a3f5b026f69a842af9d88/dataplane/app.go#L126" target="_blank" rel="noopener nofollow ugc">Azure-Samples/Hybrid-Golang-Samples/blob/0fd65d23d5efd4fb6b5a3f5b026f69a842af9d88/dataplane/app.go#L126</a></h4>

  <pre class="onebox">    <code class="lang-go">
      <ol class="start lines" start="116" style="counter-reset: li-counter 115 ;">
          <li>
          </li>
<li>	fmt.Printf("Creating storage account '%s'...\n", storageAccountName)</li>
          <li>	// Create storage account</li>
          <li>	_, errSa := hybridstorage.CreateStorageAccount(</li>
          <li>		cntx,</li>
          <li>		storageAccountClient,</li>
          <li>		storageAccountName,</li>
          <li>		resourceGroupName,</li>
          <li>		config.Location)</li>
          <li>	if errSa != nil {</li>
          <li class="selected">		log.Fatal(errSa.Error())</li>
          <li>	} else {</li>
          <li>		fmt.Printf("Successfully created storage account %s.\n", storageAccountName)</li>
          <li>	}</li>
          <li>
          </li>
<li>	fmt.Printf("Getting dataplane URL...\n")</li>
          <li>	dataplaneURL, errDP := hybridstorage.GetDataplaneURL(</li>
          <li>		cntx,</li>
          <li>		storageAccountClient,</li>
          <li>		storageEndpointSuffix,</li>
          <li>		storageAccountName,</li>
      </ol>
    </code>
  </pre>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

**Comment by TheOnlyWei** [comment, marked as answer]
<p>The issue turned out to be that I needed to obtain the last exit code from the calling application, which is PowerShell Core, and check the exit code and if it is non-zero, throw an error from that calling application in PowerShell core, otherwise Github Actions won’t register it as an error.</p>

**Reply by eugbyte** [reply]
Do you have the script for that?
